docs/extensions/readme.md

```diff
@@ -1173,3 +1175,56 @@ An object with an optional property.
   }
 }
 ```
+
+## x-ms-header-collection-prefix
+Use `x-ms-header-collection-prefix` to deal with prefixes for headers. Has different behavior if it refers to a request header or a response header:
+
+- Request header: All keys in the request headers will be prefixed with the prefix value before being sent to the service.
+- Response header: Only response headers that start with the prefix specified here will be returned to users. Additionally, the prefix will be stripped from the response header key before being returned to users
+
+Additionally, applying this extension to a schema forces the schema to become a dictionary.
+
+**Schema**: `string`. Name of the prefix you want to append / filter by. A common value for storage libraries is `x-ms-meta-`.
+
+**Request Example**:
+
+```json
+"parameters": [
+  "x-ms-meta": {
+    "type": "string",
+    "x-ms-client-name": "Metadata",
+    "x-ms-header-collection-prefix": "x-ms-meta-"
+  }
+]
+```
+This request parameter will be forced to be a dictionary schema, and all keys in this dictionary will be prefixed with `x-ms-meta-`.
```

Comment:
The `type` declared in the Swagger becomes the value type of the dict I guess?

Reply:
Can we define it for strings only? I think this is the only type currently in-use and would simplify the implementation.

Reply:
I'm following what @pakrym brought up in his [issue](https://github.com/Azure/autorest/issues/3761), where it should alwas be a string dictionary.

@pakrym should we model the value type to follow from the type listed instead of hardcoding it to string?

Reply:
If it was me I would throw an error for any usage of `x-ms-header-collection-prefix` with a non-string type to cut down on the scope of this feature.

Reply:
rip sorry @pakrym, both times my github didn't update with your answer so i just ended up repeating you

Reply:
I vote @pakrym , let's document that string is the only acceptable type, until someone prooves otherwise

Reply:
Yes - I would say string is the only option - it will be up to the end user (or maybe the SDK if they have prior knowledge) to further deserialize this content.

Reply:
Ok, will then look at implementing this change in m4 and ping go and csharp when the change is ready.